include/dmlc/parameter.h

```diff
@@ -575,12 +642,23 @@ class FieldEntryBase : public FieldAccessEntry {
       throw dmlc::ParamError(os.str());
     }
   }
-  virtual std::string GetStringValue(void *head) const {
+  bool Same(void* head, std::string const& value) const override {
+    DType old = this->Get(head);
+    DType now;
+    std::istringstream is(value);
+    is >> now;
+    // don't require = operator
+    bool is_same = std::equal(
+        reinterpret_cast<char*>(&now), reinterpret_cast<char*>(&now) + sizeof(now),
+        reinterpret_cast<char*>(&old));
```

Comment:
Are `old` and `new` same only when `old` and `new` have same set of bits? How about `old=-0.0` and `new=0.0`? What is the reason for avoiding the use of `=` operator?

Reply:
According to IEEE 754, `-0.0` and `0.0` are distinct values (with different bits) and at the same time `-0.0 == 0.0` should evaluate to `true`.